src/meta/size.rs

```diff
@@ -73,79 +57,68 @@ impl Size {
     ) -> ColoredString {
         let mut content = String::with_capacity(value_alignment + unit_alignment + 1);
 
-        let value = self.render_value();
-        let unit = self.render_unit(&flags);
+        let unit = self.get_unit(flags);
 
-        for _ in 0..(value_alignment - value.len()) {
+        let value_str = self.render_value(&unit);
+        let unit_str = Size::render_unit(&unit, &flags);
+
+        for _ in 0..(value_alignment - value_str.len()) {
             content.push(' ');
         }
 
-        content += &self.render_value();
-        if flags.size != SizeFlag::Short {
+        content += &self.render_value(&unit);
+        if flags.size == SizeFlag::Default {
             content.push(' ');
         }
-        content += &self.render_unit(&flags);
+        content += &Size::render_unit(&unit, &flags);
 
-        for _ in 0..(unit_alignment - unit.len()) {
+        for _ in 0..(unit_alignment - unit_str.len()) {
             content.push(' ');
         }
 
-        self.paint(colors, content)
+        self.paint(&unit, colors, content)
     }
 
-    fn paint(&self, colors: &Colors, content: String) -> ColoredString {
-        if self.unit == Unit::None {
+    fn paint(&self, unit: &Unit, colors: &Colors, content: String) -> ColoredString {
+        if unit == &Unit::None {
             colors.colorize(content, &Elem::NonFile)
-        } else if self.unit == Unit::Byte || self.unit == Unit::Kilo {
```

Comment:
I think you removed this accidentally

Reply:
Oh how did that happen... Great catch, thanks.
Rebased and fixed the first commit.